onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -369,6 +483,97 @@ void RegisterExecutionProviders(InferenceSession* sess, const std::vector<std::s
   }
 }
 
+
+/*
+ * Register execution provider with options.
+ *
+ * (note: currently only cuda EP supports this feature and rest of EPs use default options) 
+ */
+void RegisterExecutionProvidersWithOptions(InferenceSession* sess, const std::vector<std::string>& provider_types, ProviderOptionsMap& provider_options_map) {
+  PYBIND_UNREFERENCED_PARAMETER(provider_options_map);
+
+  for (const std::string& type : provider_types) {
+    if (type == kCpuExecutionProvider) {
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_CPU(sess->GetSessionOptions().enable_cpu_mem_arena));
+    } else if (type == kTensorrtExecutionProvider) {
+#ifdef USE_TENSORRT
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_Tensorrt(0));
+#endif
+    } else if (type == kMIGraphXExecutionProvider) {
+#ifdef USE_MIGRAPHX
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_MIGraphX(0));
+#endif
+    } else if (type == kCudaExecutionProvider) {
+#ifdef USE_CUDA
+      onnxruntime::CudaProviderOptions cuda_provider_options;
+
+      auto it = provider_options_map.find(type);
+      if (it != provider_options_map.end()) {
+        UpdateCudaProviderOptions(cuda_provider_options, it->second); 
```

Comment:
You already have the session object, so you should use the session logger instead of the default logger in the UpdateCudaProviderOptions function and its callees. I feel we shouldn't use the default logger unless necessary because we just had an regression in 1.3 because of that.

Reply:
That's a good suggestion. 
But for session logger, I found that session_logger_ is a private member of InferenceSession class and can't be accessed from outside the class. Would you suggest to add a public method within the InferenceSession class to get session logger? So that we can use this session logger to print some useful information from pybind. 
Or you have better suggestion on how to access the session logger?
Thanks.